### PR TITLE
Card carousel

### LINKED ISSUE #134: Create a carousel component for codefeed section
Create a carousel component for codefeed section

### PR DESCRIPTION
**Pull request by rijusougata13**
Added a carousel component for codefeed section

## Motivation and Context

Fixed #134 

## Screenshots or GIF (In case of UI changes):
simplescreenrecorder-2021-06-29_19 56 37

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/components/MyFeed/index.js [hunk 1]
```diff
@@ -23,12 +26,15 @@ const MyFeed = () => {
...
+          <div item xs={12}>
+            <Carousel />
+          </div>
         </Grid>
 
-        <Grid item xs={12}>
-          {" "}
-          <ExploreOrgs />
-        </Grid>
+        {/* <Grid item xs={12} > */}
```
Comment (Niweera) [inline src/components/MyFeed/index.js:34]: Is there any reason why these lines are commented out?

Comment (rijusougata13) [reply src/components/MyFeed/index.js:34]: actually those are previous view thats why commented them out , Now everyone approved this design I will remove those comment

File: src/components/MyFeed/Carousel/index.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      opacity: 1,
+
+      transform: "scale(.6,.6) rotateY(-100deg)",
+    },
+    "100%": {
+      opacity: 1,
+      transform: "scale(1,1) rotateY(0)",
+    },
+  },
+}));
```
Comment (shehand) [inline src/components/MyFeed/Carousel/index.js:105]: Shall we move the css to a different file? It is ok to have them if they are few lines long but not this much long.

File: src/components/MyFeed/Carousel/index.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+                        zIndex: -1,
+                      }}
+                    >
+                      <CardActionArea>
+                        <CardMedia
+                          className={classes.media}
+                          alt="CodeLabz"
+                          component="img"
+                          title="CodeLabz"
+                          // image={org.org_image}
```
Comment (shehand) [inline src/components/MyFeed/Carousel/index.js:176]: If no longer needed, remove these kind of comments

File: src/config/index.js [hunk 4]
```diff
@@ -31,7 +31,9 @@ export const functions = firebase.functions();
 let firebase_messaging;
 if (firebase.messaging.isSupported()) {
   firebase_messaging = firebase.messaging();
-  firebase_messaging.usePublicVapidKey(process.env.REACT_APP_FIREBASE_FCM_VAPID_KEY);
```
Comment (shehand) [inline src/config/index.js:34]: You really need to consider this.

**Review by shehand — CHANGES_REQUESTED** [review]

File: src/components/MyFeed/Carousel/index.js (hunk 3 above)
Comment (rijusougata13) [reply src/components/MyFeed/Carousel/index.js:176]: ok i am removing unnecessary code

File: src/config/index.js (hunk 4 above)
Comment (rijusougata13) [reply src/config/index.js:34]: @shehand sorry for this problem, but couldn't fixed this issue in this PR , But after #141  this problem solved . I tried the fix mentioned in the stackoverflow but it didn't worked on those pr which are already created .  That fix is working on new branch I am creating. This firebase config will not bother anymore in future PRs.

File: src/components/MyFeed/index.js (hunk 1 above)
Comment (rijusougata13) [reply src/components/MyFeed/index.js]: fixed

File: src/components/MyFeed/Carousel/index.js (hunk 2 above)
Comment (rijusougata13) [reply src/components/MyFeed/Carousel/index.js]: fixed

File: src/components/MyFeed/Carousel/index.js (hunk 3 above)
Comment (rijusougata13) [reply src/components/MyFeed/Carousel/index.js]: fixed

**Comment by rijusougata13** [comment]
sorry @shehand  all the prs before #141  have same config file gitignore issue . I have tried but that fix breaking the code . So couldn't applied those fix on those pr which are before #141 . But after #141 pr the problem solved.

File: src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js [hunk 5]
```diff
@@ -39,22 +47,43 @@ const EditProfileDetailsModal = ({ profileData, modelCloseCallback }) => {
...
+  const [description, setDescription] = useState(
+    getData(profileData.description)
+  );
+  const [descriptionValidateError, setDescriptionValidateError] = useState(
+    false
+  );
+  const [
+    descriptionValidateErrorMessage,
+    setDescriptionValidateErrorMessage,
+  ] = useState("");
```
Comment (shehand) [inline src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js:63]: Could you please revert this back to the original. Better to have these in a single line

File: src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js [hunk 6]
```diff
@@ -39,22 +47,43 @@ const EditProfileDetailsModal = ({ profileData, modelCloseCallback }) => {
...
+  const [
+    linkedinValidateErrorMessage,
+    setLinkedinValidateErrorMessage,
+  ] = useState("");
   const [github, setGithub] = useState(getData(profileData.link_github));
   const [githubValidateError, setGithubValidateError] = useState(false);
-  const [githubValidateErrorMessage, setGithubValidateErrorMessage] = useState("");
+  const [githubValidateErrorMessage, setGithubValidateErrorMessage] = useState(
+    ""
+  );
```
Comment (shehand) [inline src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js:86]: same as these

**Review by shehand — APPROVED** [review]

File: src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js (hunk 6 above)
Comment (rijusougata13) [reply src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js:86]: Ok I am reverting this to one line.

**Comment by thisuraseniya** [comment]
Hi @rijusougata13, please fix the merge conflicts and commit :D

**Comment by rijusougata13** [comment]
ok i am solving this conflict

**Comment by rijusougata13** [comment]
@thisuraseniya  solved the conflict

**Review by Niweera — APPROVED** [review]

**Pull request merged**